api/desecapi/views/records.py

```diff
@@ -99,6 +99,10 @@ def update(self, request, *args, **kwargs):
         return response
 
     def perform_destroy(self, instance):
+        # Disallow modification of apex NS RRset for locally registrable domains
+        if instance.type == "NS" and self.domain.is_locally_registrable:
+            if instance.subname == "":
+                raise ValidationError("Cannot modify NS records for this domain.")
```

Comment:
This validation is in the view layer, whereas other validations in this PR are in the serializer layer. For better maintenance, related validations should be located in the same layer.

Reply:
Deletion does not go through the serializer in DRF, except on bulk endpoints. That's why it's different (not our decision).